homeassistant/components/water_heater/__init__.py

```diff
@@ -136,6 +146,9 @@ def precision(self):
     def state_attributes(self):
         """Return the optional state attributes."""
         data = {
+            ATTR_CURRENT_TEMPERATURE: show_temp(
+                self.hass, self.current_temperature, self.temperature_unit,
+                self.precision),
```

Comment:
I can imagine some devices don't have a current temperature, so should this be optional? My device (`evohome`) has current temperature, but no target temperature!

There is no consensus on this on the architectural discussion.

Reply:
My device doesn't support current temprature so it just remains unfilled. What is the proper way to make this optional? And still make it available.

Reply:
Just have a look how it is done fro `SUPPORT_OPERATION_MODE`